src/material-experimental/mdc-slider/slider-thumb.ts

```diff
@@ -87,10 +105,44 @@ export interface MatSliderDragEvent {
   /** Indicates which slider thumb this input corresponds to. */
   thumb: Thumb;
 
+  /** Whether the value of this slider thumb input has been set. */
+  _initialized: boolean = false;
+
   constructor(
     @Inject(DOCUMENT) private readonly _document: Document,
     readonly _elementRef: ElementRef<HTMLInputElement>,
-    ) {}
+    private readonly _slider: MatSlider,
+    ) {
+      // Initializing the min and max in the constructor guarantees that they will be
+      // defined by the time the value gets set. If the range is not defined before we
+      // try to set the value, we can run into the issue where the value is outside of
+      // the default range and get capped to the default min or max.
+      this._elementRef.nativeElement.min = this._slider.min.toString();
+      this._elementRef.nativeElement.max = this._slider.max.toString();
+    }
+
+  /**
+   * Sets up the initial state of the slider thumb input.
+   *
+   * This is needed because the slider thumb input is passed in via `ng-content`,
+   * and therefore has no way of knowing which slider thumb it correspond to.
+   */
+  _init(thumb: Thumb): void {
+    this.thumb = thumb;
+
+    // If the value has not been initialized (i.e. no value was provided from
+    // the user), determine the default value for the slider based on the given thumb.
+    if (!this._initialized) {
+      this.value = (this._slider._isRange() && thumb === Thumb.END)
+        ? this._slider.max
+        : this._slider.min;
+    }
+
+    // Setup for the MDC foundation.
+    if (this._slider.disabled) {
+      this._elementRef.nativeElement.disabled = true;
+    }
```

Comment:
Similar question here- why does this need to be set in the DOM like this instead of relying on internal component state?